bin/fetch-recent-miner-measurements.js

```diff
@@ -0,0 +1,214 @@
+// dotenv must be imported before importing anything else
+import 'dotenv/config'
+
+import { Point } from '@influxdata/influxdb-client'
+import * as Sentry from '@sentry/node'
+import createDebug from 'debug'
+import fs from 'node:fs'
+import { mkdir, readFile, writeFile } from 'node:fs/promises'
+import os from 'node:os'
+import path from 'node:path'
+import pMap from 'p-map'
+import { IE_CONTRACT_ADDRESS } from '../lib/config.js'
+import { createMeridianContract } from '../lib/ie-contract.js'
+import { fetchMeasurements, preprocess } from '../lib/preprocess.js'
+import { RoundData } from '../lib/round.js'
+
+const {
+  STORE_ALL = false
+} = process.env
+
+Sentry.init({
+  dsn: 'https://d0651617f9690c7e9421ab9c949d67a4@o1408530.ingest.sentry.io/4505906069766144',
+  environment: process.env.SENTRY_ENVIRONMENT || 'dry-run',
+  // Performance Monitoring
+  tracesSampleRate: 0.1 // Capture 10% of the transactions
+})
+
+const debug = createDebug('spark:bin')
+
+const cacheDir = path.resolve('.cache')
+await mkdir(cacheDir, { recursive: true })
+
+const [nodePath, selfPath, ...args] = process.argv
+if (args.length === 0 || !args[0].startsWith('0x')) {
+  args.unshift(IE_CONTRACT_ADDRESS)
+}
+const [contractAddress, minerId, blocksToQuery] = args
+
+const USAGE = `
+Usage:
+  ${nodePath} ${selfPath} [contract-address] minerId [blocksToQuery]
+`
+
+if (!contractAddress) {
+  console.error('Missing required argument: contractAddress')
+  console.error(USAGE)
+  process.exit(1)
+}
+
```

Comment:
```suggestion
```

With the logic above I believe this is dead code

Reply:
You can still provide an empty string in the CLI arg, right?

Reply:
You're right 👍 

```
$ node -pe "console.log(process.argv)" ""
[ '/Users/julian/.nvm/versions/node/v20.12.1/bin/node', '' ]
```

Reply:
Well, we detect the empty string and insert the contract address. Then we interpret the empty string as a miner id.

```
❯ node bin/fetch-recent-miner-measurements.js '' f010479 
Selected JSON-RPC endpoint https://api.node.glif.io/rpc/v0
Missing required argument: minerId

Usage:
  /usr/local/bin/node /Users/bajtos/src/spark/evaluate/bin/fetch-recent-miner-measurements.js [contract-address] minerId [blocksToQuery]
```

I think this is an edge case not worth spending our time on, but let me know if you disagree.

Reply:
I understand, let's get back to this when it becomes a problem?